src/averageAges.js

```diff
@@ -1,64 +1,44 @@
 
-/**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
- * @param {object[]} people
- * @param {number} century - optional
- *
- * @return {number}
- */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const peopleFiltered = people.filter((person) => century
+    ? person.sex === 'm' && Math.ceil(person.died / 100) === century
+    : person.sex === 'm');
+
+  return (peopleFiltered.reduce((accumulator, currentValue) =>
+    (accumulator + (currentValue.died - currentValue.born)), 0))
+    / peopleFiltered.length;
 }
 
-/**
- * Implement calculateWomenAverageAge function
- *
- * Function returns average age of women in array. If `withChildren` is
- * specified then function calculates average age only for women with children
- *
- * Hint: To check if a woman has children you should find someone who mention
- * her as mother.
- *
- * @param {object[]} people
- * @param {boolean} withChildren - optional
- *
- * @return {number}
- */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const peopleFiltered = people.filter((person) => withChildren
+    ? people.some((child) => child.mother === person.name)
+    : person.sex === 'f');
+
+  return (peopleFiltered.reduce((accumulator, currentValue) =>
+    (accumulator + (currentValue.died - currentValue.born)), 0))
+    / peopleFiltered.length;
 }
 
-/**
- * Implement calculateAverageAgeDiff function.
- *
- * The function returns an average age difference between a child and his or her
- * mother in the array. (A mother's age at child birth)
- *
- * If `onlyWithSon` is specified then function calculates age difference only
- * for sons and their mothers.
- *
- * @param {object[]} people
- * @param {boolean} onlyWithSon - optional
- *
- * @return {number}
- */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const ageDifference = [];
+
+  const peopleFiltered = people.filter((person) => onlyWithSon
+    ? person.sex === 'm' && person.mother !== null
+    : person.mother !== null);
+
+  const motherFiltered = people.filter((person) =>
+    people.some((mother) => mother.mother === person.name));
+
+  peopleFiltered.forEach((person) =>
+    motherFiltered.find((mother) => {
+      if (mother.name === person.mother) {
+        ageDifference.push(person.born - mother.born);
+      }
+    }));
```

Comment:
Oki, but better to use `map and return array in such format:
```
[{ died: ..., born: ... }]
```

Then you can use `calculateAverageAge` function to calculate average age.